### DISCUSSION TITLE
I'm trying to create a lexer for a new language

### DISCUSSION THREAD

**Discussion by nsgnkhibdk2cls0f** [orbitalquark/textadept#307] [Q&A] [answered]
The language uses `#~` to open and `~#` to close multi-line comment, `#` for single-line comment. 

This is what I have on the lexer file:

```
-- Comments.
local line_comment = lexer.to_eol('#', true)
local block_comment = lexer.range('#~', '~#')
lex:add_rule('comment', token(lexer.COMMENT, line_comment + block_comment))
```

Strangely enough, only single-line comment works, multi-line comment doesn't work. Please help.

**Comment by orbitalquark** [comment, marked as answer]
Block comments need to come first, then line comments. The lexer matches the first pattern it can.
